### DISCUSSION TITLE
ZIP file upload fails

### DISCUSSION THREAD

**Discussion by Vodhin** [e107inc/e107#5002] [Q&A Plugins] [answered]
I hope someone can help: I have developed a number of websites using this e107 since 2005 but I haven't published any of my plugins before. Now I am nearing a point in my Real Estate Listings plugin development where I'd like to get feedback: a production version can be seen [here](https://www.sandpiperhome.org) .  In the past (v7.4) I was able to zip up a plugin and successfully upload it to another site but now I get this (using e107 Version 2.3.2):
 
```
ERROR
Couldn't detect the root folder in the zip.
There was a problem extracting the .zip file to your plugin directory.
```

I am using the native WIN10 _SendTo->Compressed File_ function to generate the zip from a local copy of the plugin folder. If I upload the uncompressed folder via FTP everything is fine. I've dug around both here and in the e107 source code but have not found any clue as to what happened: I've made sure that PHPini settings are OK and e107 filetypes allow the zip. 

 I will attempt to add the zip file here (for examination of the ZIP file, the plugin needs a lot of cleanup).

[estate.zip](https://github.com/e107inc/e107/files/11615850/estate.zip)

**Comment by Jimmi08** [comment]
Thanks for sharing. I am not aware if this is option (they supported to upload into e107.org and install from it) .  I always use FTP and manual upload to server.

**Reply by Vodhin** [reply]
Thank you. I just want to check the ZIP file format/structure to make sure others can try this plugin regardless of how it is added to their website. There is an upload option in the Plugin Manager, which is what I was using for this test. Not very important at the moment, I guess. Right now I'm fighting another issue with the e_frontpage file breaking the frontpage prefs.

**Reply by Jimmi08** [reply]
I was looking for similar plugin from my beginning with e107. Now I use WordPress solution, I don't need it anymore, but if you upload it as your repo, I can help.

**Reply by Vodhin** [reply]
I'm new to Github and I've managed to create a repository and have uploaded the main files. I cannot for the life of me figure out how to create the folders for the other files.

**Comment by tgtje** [comment]
Well lets try a help here @Vodhin 
on pc) go into github (logged in at account) go to repositories

 top right > NEW > click it
fill in what you like to use ( do mind public (to all) or private (needs granted access for others).
yes/no read me etc..
bottom > create
see the images
on second image you see the upload files (easiest) opens a new window
where you can upload a full folder or single file by drag and drop OR manual
DO NOTE!!!! do NOT try to upload more than 100 files in 1 go ( =folder/subs and internal files).
If you have a plain plugin folder ready and less than 100 files in total : just drop it.
add some text for the commit (below) and hit the commit changes button.

git will keep the structure.

If your repo has more than 100 files you have to do a partial upload.
just needs a little attention that the 'later parts' need some extra work.
eg split a folder in parts, and each time select the main folder (already been uploaded) on github 
where the other files belong into)
the method is the same.
example if you have a folder with 10 subs split main into half (do store the others somewhere else for now).
upload the first bunch. As structure is maintained, upload first half.
When done, go to the main repo you just uploaded, click the foldername you just uploaded (eg = container).
Now again in THIS selected folder  make a new upload and add the rest.
When completed the github repo should dispaly 1 folder with all the subs inside.
Side note : delete the repo ? click on top on Settings,agree on some things and delete (account pw needed).

![cr](https://github.com/e107inc/e107/assets/1811857/183ec6a6-1c98-4438-9c4a-66ac4932de90)


![cr2](https://github.com/e107inc/e107/assets/1811857/5aae1844-6007-42b2-aeac-ea12ddbc2bc1)
 ask iff any question..

result

![cr4](https://github.com/e107inc/e107/assets/1811857/71103112-2301-4368-904f-e70e52d88d4e)

**Reply by Vodhin** [reply]
Thank you. Turns out I had already spent some time figuring all that out before discovering your reply (Murphy strikes again). I didn't know about the 100 file limit, though - thanks for that nugget, seems my first repository was under that limit and everything uploaded fine.

**Comment by Jimmi08** [comment, marked as answer]
@Vodhin 
your frontpage is not working because you are using e107::url() without addon e_url.php - it is for pretty urls.
Correct path for me was:  
`$config['page'][0] = array('page' => e_PLUGIN_ABS . 'estate/listings.php' , 'title' => EST_GEN_LISTINGS); `

Always use full path there (if above fails for you, try e_HTTP, SITEURL or something)

**Reply by Vodhin** [reply]
Thank you for the tip. I did manage to get this working in the e_frontpage.php
`$config['page'][0] = array('page' => e107::url('estate', 'listings'), 'title' => EST_GEN_LISTINGS); `

and added the following to a new e_url.php

`$config['listings']=array('alias'=>EST_SEF_LISTINGS,'regex'=>'^{alias}/?','sef'=>'{alias}','redirect'=>{e_PLUGIN}estate/listings.php');`
